ode-wrapper/src/main/java/com/trihydro/odewrapper/helpers/SetItisCodes.java

```diff
@@ -163,24 +162,56 @@ public List<String> setItisCodes(WydotTim wydotTim) {
                 items.add(code.getItisCode().toString());
         }
 
-    return items;
-  }
+        return items;
+    }
+
+    public List<String> setItisCodesVsl(WydotTimVsl wydotTim) {
+
+        List<String> items = new ArrayList<String>();
+
+        List<ItisCode> allItisCodes = getItisCodes();
+
+        // add advisory code first if advisory speed limit
+        wydotTim.getItisCodes().stream().filter(x -> x.contains("7712")).findFirst()
+                .ifPresent(items::add);
+
+        // speed limit itis code
+        allItisCodes.stream().filter(x -> x.getDescription().equals("speed limit")).findFirst().ifPresent(mph -> items.add(mph.getItisCode().toString()));
 
-  public List<String> setItisCodesBowr(WydotTimBowr tim) throws WeightNotSupportedException {
-    List<String> itisCodes = new ArrayList<>();
+        // add reduced code next if reduced speed
+        wydotTim.getItisCodes().stream().filter(x -> x.contains("12302")).findFirst()
+                .ifPresent(items::add);
 
-    int weightInPounds = tim.getData();
+        // number e.g 50, convert to ITIS code
+        items.add(Integer.toString(
+                Optional.ofNullable(wydotTim.getSpeed()).orElse(0) + 12544
+        ));

```

Comment:
Question (blocking): Some clarification as to why 12544 is added to the number would be helpful here. For instance, if the number is 50, what does the output ITIS code (12594) correspond to? Is there a predictable pattern to the sequence of numbers or are there any big jumps we should know about? (i.e. 50 -> 60 instead of 50 -> 51)

Reply:
That sounds right then. Yes please! 💯